CONTRIBUTING.md

```diff
@@ -1,7 +1,122 @@
-# Contributing to a Rust Foundation Project
+# Contributing to the coding guidelines
 
-Thank you for your interest in contributing to this Rust Foundation project. 
-We are happy and excited to review and accept your pull requests.
+## Table of Contents
+
+- [Contributing to the coding guidelines](#contributing-to-the-coding-guidelines)
+  - [Table of Contents](#table-of-contents)
+  - [Contribution Workflow](#contribution-workflow)
+    - [Note on Chapter Layout](#note-on-chapter-layout)
+    - [0) Bring up the idea for discussion](#0-bring-up-the-idea-for-discussion)
```

Comment:
Might be helpful to also include the "optional" marker here. Can be done in a follow-up.